core/src/main/java/org/infinispan/util/concurrent/jdk8backported/StrippedConcurrentLinkedDeque.java

```diff
@@ -0,0 +1,723 @@
+package org.infinispan.util.concurrent.jdk8backported;
```

Comment:
Just to ease review, this class is pretty much copy pasted from ConcurrentLinkedDeque in the jdk with a ton of methods removed.  

There are a few extra methods that were added to expose the underlying Nodes to allow for O(1) removal times that I can annotate here (normally removal requires full iteration).